pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/TopicTransactionBufferRecoverTest.java

```diff
@@ -550,7 +558,7 @@ private void checkSnapshotCount(TopicName topicName, boolean hasSnapshot,
         CompletableFuture<Long> compactionFuture = (CompletableFuture<Long>) field.get(persistentTopic);
         Awaitility.await().untilAsserted(() -> assertTrue(compactionFuture.isDone()));
 
-        Reader<TransactionBufferSnapshot> reader = pulsarClient.newReader(Schema.AVRO(TransactionBufferSnapshot.class))
+        Reader<GenericRecord> reader = pulsarClient.newReader(Schema.AUTO_CONSUME())
```

Comment:
we also need to check snapshot index topic right?

Reply:
Add the test logic in `SegmentAbortedTxnProcessorTest::testClearSnapshotSegments`.